File: frontend/src/api/supplierApi.js
import apiClient from "./apiClient";

// ---------------------------------------------------------
// Get All Suppliers
// ---------------------------------------------------------

export const getSuppliers = async () => {

    const response = await apiClient.get(
        "/suppliers"
    );

    return response.data;

};

// ---------------------------------------------------------
// Get Supplier By Id
// ---------------------------------------------------------

export const getSupplier = async (
    supplierId
) => {

    const response = await apiClient.get(
        `/suppliers/${supplierId}`
    );

    return response.data;

};

// ---------------------------------------------------------
// Create Supplier
// ---------------------------------------------------------

export const createSupplier = async (
    supplier
) => {

    const response = await apiClient.post(
        "/suppliers",
        supplier
    );

    return response.data;

};

// ---------------------------------------------------------
// Update Supplier
// ---------------------------------------------------------

export const updateSupplier = async (
    supplierId,
    supplier
) => {

    const response = await apiClient.put(
        `/suppliers/${supplierId}`,
        supplier
    );

    return response.data;

};

// ---------------------------------------------------------
// Delete Supplier
// ---------------------------------------------------------

export const deleteSupplier = async (
    supplierId
) => {

    const response = await apiClient.delete(
        `/suppliers/${supplierId}`
    );

    return response.data;

};

File: api/supplier_api.py
from fastapi import APIRouter, HTTPException

from schemas.supplier_schema import (
    SupplierCreate,
    SupplierUpdate
)

from services.supplier_service import SupplierService

router = APIRouter()

service = SupplierService()


@router.get("")
def get_all():

    return service.get_all()


@router.get("/{supplier_id}")
def get_supplier(supplier_id: int):

    try:

        return service.get_by_id(supplier_id)

    except ValueError as ex:

        raise HTTPException(
            status_code=404,
            detail=str(ex)
        )


@router.post("")
def create_supplier(request: SupplierCreate):

    try:

        return service.create(
            request.model_dump()
        )

    except ValueError as ex:

        raise HTTPException(
            status_code=409,
            detail=str(ex)
        )


@router.put("/{supplier_id}")
def update_supplier(
    supplier_id: int,
    request: SupplierUpdate
):

    try:

        return service.update(
            supplier_id,
            request.model_dump()
        )

    except ValueError as ex:

        raise HTTPException(
            status_code=404,
            detail=str(ex)
        )


@router.delete("/{supplier_id}")
def delete_supplier(supplier_id: int):

    try:

        service.delete(supplier_id)

        return {
            "message": "Supplier deleted successfully."
        }

    except ValueError as ex:

        raise HTTPException(
            status_code=404,
            detail=str(ex)
        )